lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/IntBitUnpackers.java

```diff
@@ -0,0 +1,1458 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader.decoders;
+
+import io.trino.parquet.reader.SimpleSliceInputStream;
+
+public final class IntBitUnpackers
```

Comment:
doesn't all the unpackers implementations create megamorphism? How big are inputs streams processed by unpackers?

Reply:
Those unpackers work on batches of the same size as decoders. Usually several thousands of values.

Reply:
Minimum number of values in bitpacked encoding is 8. The length of the run depends on the data. If a value repeats >= 8 times, then we get an RLE run, otherwise it's a bitpacked run.
A few files I looked at had runs of around 500.
In any case, I've removed the interface and replaced it with an explicit switch case. JMH results even with profile pollution are vastly better than the parquet-mr decoder.

Reply:
Is the case clause better? It's potentially 31 comparisons vs one array lookup

Reply:
The switch case turned out to be slightly better in JMH but a lot worse in TPC runs, so I guess we go back to the original approach.
In theory JIT should optimise a switch case into a lookup table where it's beneficial.